### PR TITLE
Routine to properly print and save config data

### PR DESCRIPTION
**Pull request by wllacer**
As outlined at issue #873 (in comments) there were some issues with the printing and saving of the config data ( aka *archinstall.arguments*) as some of them were duplicated.
We have prepared a routine which print and/or saves the data in json format, with the desired separations (general, disk layout and credentials -which aren't printed-)
`def output_configs(show :bool = True, save :bool = True):`
it's use is simple
```python
from archinstall.examples.commons import output_configs
....
`output_configs(show=False if archinstall.arguments.get('silent') else True)
```
What can be more controversial is where i did put it. As it was a function which is (and could be by others) easyly reused, i've put them in a module (called *commons*) at the *examples* directory. Why there?
It is a function which accesses the whole *arguments* dictionary, and uses a number of archinstall.* calls. The second can be solved, but the first implies that it can not go into most of the modules at *lib* directory without the risk of creating a circular dependency, or needing an explicit parameter for the dictionary, in what is a very specific function ONLY for this structure.
And is a function which a script developer would have some use to customize.

I have also with the new menu some similar issues (where to put code which ought to use the archinstall.* notation, and are third party developer customizable). They should go into a module which can do an ìmport archinstall` and can, with reason and some easy be user customizable. The obvious place in this case, now is at *examples*; but i don't know if in other situation (a great deal of code menu related) it would not be a better idea to have a sibling directory to  *lib* as the default, from which developers can overload in their own structure.

### CODE REVIEW

File: examples/guided.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import json
 import logging
 import os
 import time
 
 import archinstall
+from archinstall.examples.commons import output_configs
```
Comment (Torxed) [inline examples/guided.py:6]: I would say cross-importing examples is strictly forbidden.
The examples should only serve as an example of *"how would you use archinstall"* type of thing.

So I would say this is a blocker and I won't be able to merge this PR sadly.
If we move the logic of saving configuration to `lib/environment/configuration.py` or similar, that would work. We could also move the loading of configuration to that file.

I don't mind making those changes if you wish, but feel free to change if you have time and/or want to :)

**Review by Torxed — CHANGES_REQUESTED** [review]

**Comment by wllacer** [comment]
I knew this was about to happen ...
I'll try to get first an *lib* allocatable version of the output routine.
There is too much need for It.
And then we can look for a place for code like this ...

**Comment by Torxed** [comment]
Sounds good, great work tho! I'll wait for new commits to this PR before merging, you do not need to open a new one. And no rush, I know you're busy at the moment :)

**Comment by wllacer** [comment]
A bit late tonight ;-) It was not that complex to change, and got a quiet moment. With a litte effort and some more parameters, it could be a general solution for split printing dictionaries ... The location is just provisional, but not a roadblock anymore.
BTW. Have a look at the elements which go to user_credentials.json. I'm missing the `!root_password` argument. But it wasn't there before either.

**Comment by Torxed** [comment]
`!root-password` is omitted, and only inserted if the user chose to set one during earlier steps. Maybe we should localize all those arguments close to each other so that it's easier to find the logic. Good work on this tho, merged! : )

**Pull request merged**